thali/thalireplicationmanager.js

```diff
@@ -188,15 +202,62 @@ function syncPeer(peer) {
 
 /* Mux Layer */
 
+function restartMuxServerBridge() {
+  this.stop();
+
+  this.on('stopped', function () {
+    this.start(this._deviceName, this._port, this._dbName);
+  }.bind(this));
+}
+
 function muxServerBridge(tcpEndpointServerPort) {
   var serverPlex = multiplex({}, function(stream, id) {
     var clientSocket = net.createConnection({port: tcpEndpointServerPort});
     stream.pipe(clientSocket).pipe(stream);
   });
 
-  return net.createServer(function(incomingClientSocket) {
+  var server = net.createServer(function(incomingClientSocket) {
+
+    incomingClientSocket.on('error', function (err) {
+      console.log('incoming client socket error %s', err);
+      try {
+        serverPlex.destroy();
+        server.close();
+      } catch (e) {
+        console.log('failed to clean up server and serverPlex');
+      }
+
+      restartMuxServerBridge.call(this);
```

Comment:
Fixed with a flag to determine whether the cleanup has been called before so it can only be called once.